flow/src/main/java/flow/History.java

```diff
@@ -121,20 +120,26 @@ public int size() {
 
   public <T> T top() {
     //noinspection unchecked
-    return (T) history.peek().state;
+    return (T) peek(0);
+  }
+
+  public <T> T peek(int index) {
```

Comment:
Meh, then History has top() / peek(int), and builder would have peek() / peek(int). Seems like we need to make up our minds. I'll leave Builder alone for now.